lib/knapsack_pro/client/connection.rb

```diff
@@ -59,18 +59,15 @@ def request_body
         request_hash.to_json
       end
 
-      def test_suite_token
-        KnapsackPro::Config::Env.test_suite_token
-      end
-
       def json_headers
         {
           'Content-Type' => 'application/json',
           'Accept' => 'application/json',
           'KNAPSACK-PRO-CLIENT-NAME' => client_name,
           'KNAPSACK-PRO-CLIENT-VERSION' => KnapsackPro::VERSION,
-          'KNAPSACK-PRO-TEST-SUITE-TOKEN' => test_suite_token,
-        }
+          'KNAPSACK-PRO-TEST-SUITE-TOKEN' => KnapsackPro::Config::Env.test_suite_token,
+          'KNAPSACK-PRO-CI-PROVIDER' => KnapsackPro::Config::Env.ci_provider,
+        }.compact
```

Comment:
Since you added `compact`. Suppose someone forgot to set the API token. Will the API still return a meaningful error?

Reply:
@ArturT I removed the token from the script and I got this:

```bash
bin/knapsack_pro_rspec  
rake aborted!
Missing environment variable KNAPSACK_PRO_TEST_SUITE_TOKEN. You should set environment variable like KNAPSACK_PRO_TEST_SUITE_TOKEN_RSPEC (note there is suffix _RSPEC 
at the end). knapsack_pro gem will set KNAPSACK_PRO_TEST_SUITE_TOKEN based on KNAPSACK_PRO_TEST_SUITE_TOKEN_RSPEC value. If you use other test runner than RSpec then 
use proper suffix.
```

It works because we first build the hash calling `KnapsackPro::Config::Env.test_suite_token` and, only then, `compact` it.

Reply:
ok. Thanks for checking.